CMakeLists.txt

```diff
@@ -146,3 +154,58 @@ target_link_libraries(test_kmeans PRIVATE lib_kmeans fortuno_wrapper common_flag
 add_test(NAME test_assign_points_to_centroids COMMAND test_kmeans "Assign points to centroids")
 add_test(NAME test_update_centroids_no_movement COMMAND test_kmeans "Update centroids - no change")
 add_test(NAME test_points_are_converged COMMAND test_kmeans "Difference in two sets of points")
+
+# Installation
+# --------------------------------
+
+# Library
+install(
+   TARGETS lib_kmeans
+   EXPORT KmeansTargets
+   LIBRARY
+     DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+   ARCHIVE
+     DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+ )
+
+install(EXPORT KmeansTargets
+   FILE KmeansTargets.cmake
+   NAMESPACE kmeans::
+   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/kmeans"
+   )
+
+# Module files.
+install(
+   DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}
+   DESTINATION "."  # Will automatically get placed in "${CMAKE_INSTALL_LIBDIR}/modules"
+ )
```

Comment:
I note that no modules currently install, perhaps since I changed lib_kmeans src to PRIVATE?